lnwire/message_test.go

```diff
@@ -0,0 +1,844 @@
+package lnwire_test
+
+import (
+	"bytes"
+	"encoding/binary"
+	"image/color"
+	"io"
+	"math"
+	"math/big"
+	"math/rand"
+	"net"
+	"sync"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/tor"
+	"github.com/stretchr/testify/require"
+)
+
+const deliveryAddressMaxSize = 34
+const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+var (
+	testSig = &btcec.Signature{
+		R: new(big.Int),
+		S: new(big.Int),
+	}
+	testNodeSig, _ = lnwire.NewSigFromSignature(testSig)
+
+	testNumExtraBytes = 1000
+	testNumSigs       = 100
+	testNumChanIDs    = 1000
+	buffer            = make([]byte, 0, lnwire.MaxSliceLength)
+
+	bufPool = sync.Pool{
+		New: func() interface{} {
+			return bytes.NewBuffer(buffer)
```

Comment:
Hmm, does this end _copying_ the buffer each time `New` is called, or create a new one in place?

Reply:
It's arbitrary. From the [godoc](https://golang.org/pkg/sync/#Pool.Get),
> Get selects an arbitrary item from the Pool, removes it from the Pool, and returns it to the caller. Get may choose to ignore the pool and treat it as empty. Callers should not assume any relation between values passed to Put and the values returned by Get.

I tested locally. If I run the test exactly once (`b.N` = 1), it will create two buffers and keeps iterating between the two.